pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -223,6 +225,16 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation, errorContext *er
 			Fn:           flow.TaskFn(botanist.WaitUntilKubeAPIServerReady).DoIf(cleanupShootResources),
 			Dependencies: flow.NewTaskIDs(wakeUpControlPlane),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying internal domain DNS record",
+			Fn:           flow.TaskFn(botanist.DeployInternalDomainDNSRecord).DoIf(dnsEnabled && managedInternalDNS && cleanupShootResources),
+			Dependencies: flow.NewTaskIDs(wakeUpControlPlane),
+		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying external domain DNS record",
+			Fn:           flow.TaskFn(botanist.DeployExternalDomainDNSRecord).DoIf(dnsEnabled && managedExternalDNS && cleanupShootResources),
+			Dependencies: flow.NewTaskIDs(wakeUpControlPlane),
+		})
 		initializeShootClients = g.Add(flow.Task{
 			Name:         "Initializing connection to Shoot",
 			Fn:           flow.SimpleTaskFn(botanist.InitializeShootClients).DoIf(cleanupShootResources).RetryUntilTimeout(defaultInterval, 2*time.Minute),
```

Comment:
I think this task should be dependent on the deployment of the internal DNS record.

Reply:
As discussed, I added it to the deletion as well as to the reconcile flow.